.github/workflows/validation.yml

```diff
@@ -0,0 +1,54 @@
+on:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - master
+    tags:
+      - 'v**'
+
+name: validation
+jobs:
+  validate:
+    name: Linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@master
```

Comment:
I guess GH knows to check out the correct ref? Since it says `master` here..

Reply:
@master is the version of "actions/checkout" action, not ours. It's smart about which one to checkout (pull request, or pull...)

https://github.com/actions/checkout#usage

I must admit github has done a great job with the actions stuff :)